src/css-utils.js

```diff
@@ -90,34 +47,25 @@ CSSParserUtils.SEMICOLON                                 = -2;
 * parsing in the downstream. Please be noted that &gt; itself will cause parsing error
 * no matter decode or not if it does not belong to the STRING token.
 *
-* These function decode more than we need to prevent CSS parsing error.
+* These function decode did slightly more than html decoding, we replace \r\n\f\t with space for
+* secure-handlebars for filter judgement without considering those chars.
 *
 * Reference:
 * http://www.w3.org/TR/css-style-attr/
 *
 */
-// TODO: use client side for html decode in dist/*.js version
-CSSParserUtils.reCharReferenceDecode = /&([a-z]{2,};?)|&#0*(x[a-f0-9]+|[0-9]+);?/ig;
+CSSParserUtils.reHtmlDecode = /(\t|\r|\n|\f)/g;
+CSSParserUtils.HtmlDecodeReplacement = {
+    '\t'         : ' ', // 0x09
+    '\r'         : ' ', // 0x0A
+    '\n'         : ' ', // 0x0C
+    '\f'         : ' ', // 0x0D
+};
 CSSParserUtils.htmlStyleAttributeValueEntitiesDecode = function(str) {
-    /* html decode the str before CSS parsing,
-       we follow the parsing order of the browser */
-     
-    // TODO: combine the following code with the html entities branch later.
-    // and we did slightly more than html decoding, we replace \r\n\f\t with space for
-    // secure-handlebars for filter judgement without considering those chars.
-    var fromCodePoint = String.fromCodePoint;
-    str = str.replace(CSSParserUtils.reCharReferenceDecode, function(m, named, number) {
-        if (named) {
-            return m;
-        }
-
-        var i = parseInt(number[0] <= '9' ? number : '0' + number);  // parseInt('0xA0') is equiv to parseInt('A0', 16)
-        return ((i>=0 && i<=0xD7FF) || (i>=0xE000 && i<=0xFFFF) || (i>=0x10000 && i<=0x10FFFF)) ? fromCodePoint(i) : '\uFFFD';
-    });
+    str = htmlDecoder.decode(str);
     str = str.replace(CSSParserUtils.reHtmlDecode, function(m, p) {
         return CSSParserUtils.HtmlDecodeReplacement[p];
     });
-
     return str;
 };
```

Comment:
as per the offline discussion, the `\t\r\n\f` is not needed. Hence, things will become:
`CSSParserUtils.htmlStyleAttributeValueEntitiesDecode = htmlDecoder.decode;`

pls eliminate the dead code.